### ISSUE TITLE
R code generation

### ISSUE THREAD

**Issue by nicolaskruchten** [plotly/dash#445]
`generate_classes` should generate R functions as well as Python classes.

**Comment by rpkyle** [comment]
PR #483 includes `_r_components_generation.py`, which is the analogue of `_py_components_generation.py`. Once this PR is merged into Dash, it will be possible to generate R functions for components as seamlessly as we generate Python classes.

The proposed changes also automate the R package build process, enabling component developers to support both Dash frontends without having to launch R for this step.
